File: solutions/python/Binary Tree Right Side View.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root==None:
            return []
        res=[]
        queue=[root]
        while queue:
            size=len(queue)
            for i in range(0,size):
                current=queue.pop(0)
                if i==size-1:
                    res.append(current.val)
                if current.left:
                    queue.append(current.left)
                if current.right:
                    queue.append(current.right)
        return res

File: solutions/Javascript/Binary Tree Right Side View.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

var rightSideView = function (root) {
  let queue = [root],
    res = [];
  if (!root) {
    return [];
  }
  while (queue.length > 0) {
    let size = queue.length;
    for (let i = 0; i < size; i++) {
      let current = queue.shift();
      if (i === size - 1) {
        res.push(current.val);
      }
      if (current.left) {
        queue.push(current.left);
      }
      if (current.right) {
        queue.push(current.right);
      }
    }
  }
  return res;
};
